contracts/BurningManager.sol

```diff
@@ -119,6 +126,10 @@ contract BurningManager is Initializable, AccessControlUpgradeable {
         userVars[user][USERVAR_SOUL_SUPPLY] += soulAmount;
     }
 
+    function burnSoul(address user, uint256 soulAmount) external restricted hasEnoughSoul(user, soulAmount) {
+        userVars[user][USERVAR_SOUL_SUPPLY] -= soulAmount;
```

Comment:
would recommend using .sub from the safeMath library as the transferSoul and upgradeCharacterWithSoul functions already do. It also reverts on going negative so the require statements were redundant there and the modifier should also be unnecessary (I would be curious to know how much contract space we save with this!)

Reply:
![image](https://user-images.githubusercontent.com/34208222/152962560-718a5863-bebf-49e4-a422-199f7968c4f3.png)
Not that much contract space ;) But a small gas reduction for that check should be welcome!